core/io/dir_access.cpp

```diff
@@ -250,6 +250,11 @@ DirAccess *DirAccess::create(AccessType p_access) {
 	DirAccess *da = create_func[p_access] ? create_func[p_access]() : nullptr;
 	if (da) {
 		da->_access_type = p_access;
+
+		// for ACCESS_RESOURCES and ACCESS_FILESYSTEM, current_dir already defaults to where game was started
+		if (p_access == ACCESS_USERDATA) {
+			da->change_dir("user://");
+		}
```

Comment:
It's probably worth doing the same for the `ACCESS_RESOURCES` as well.

By default, `DirAccess` uses process current directory, it's fine for `ACCESS_FILESYSTEM`. But current directory can be changed by calling `OS::set_cwd` or by third party code. If it's changed to be outside PCK location, `ACCESS_RESOURCES` will have the same issue.

Reply:
Ah okay, that's a good point